rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,152 @@
+ - Dashboard
+ - Visualize Editor
+ - Canvas
+
+**Screenshot mode service**
+
+Applications in Kibana should be aware of when their rendering is for a human
+user using the page interactively, and when it's not. At a high level, the application
+can support that awareness through a customized URL for screenshot reports. Canvas
+demonstrates a great implementation of this today. However, Canvas has dependencies
+on other UI plugins, and can not control how they render or what type of network requests
+they send. Those plugins, and lower-level plugins that run on every page, do not have
+a way to control themselves when the page is rendering for taking a screenshot.
+
+This RFC proposes a Screenshot Mode Service as a low-level plugin that allows
+other plugins (UI code) to make choices when the page is rendering for a screenshot.
+
+In most cases, the information coming from this service would help the UI code
+decide what should **not** be loaded: for example, the NewsFeed component and
+Telemetry code should not load itself when the page is rendering for
+screenshots.
+
+More background on how Reporting currently works, including the lifecycle of
+creating a PNG report, is here: https://github.com/elastic/kibana/issues/59396
+
+# Basic example
+
+When Kibana loads initially, there is a Newsfeed component in the UI that
+checks internally cached records to see if it must fetch the Elastic News
+Service for newer items. When the Screenshot Mode Service is implemented, the
+Newsfeed component has a source of information to check on whether or not it
+should load in the Kibana UI. If it can avoid loading, it avoids an unnecessary
+HTTP round trip, which weigh heavily on performance.
+
+# Motivation
+
+Kibana Reporting is a commercial feature in Elastic and is highly capable of
+loading the application pages and converting them into a screenshot for PNG or
+PDF export. However, the way it works has downsides with performance,
+maintainability, and expanding it as a tool that can power higher-level
+features. The solution to those downsides is to have Kibana pages themselves
+become more capable at presenting themselves for screenshot capture reports.
+With the Screenshot Mode Service available, Reporting could drop the
+task that it currently has which hurt performance: wasted rendering that is
+replaced with custom styles that make the page "reportable."
+
+The technical advantage of having such as service also leads to making Kibana
+application pages "printable", in the sense that sending the page to a printer
+for a hard copy results in something more meaningful and specialized for paper
+than today's Kibana can guarantee. This isn't a big concern for Kibana since
+there isn't the expectation to improve printing Kibana, but that technical
+direction is appropriate for improving PDF report generation.
+
+# Detailed design
+
+The Screenshot Mode Service is a callable API available as dependency for
+Kibana applications.
```

Comment:
the problem is that the dependency will need to be added to lower level plugins, not actually apps right ?
so we'll end up with possibly a lot of lower level plugins having an optional dependency on reporting

Reply:
That wouldn't be good: Reporting has a lot of dependencies on its own. But this service wouldn't be built as part of Reporting. It would be a new plugin, which makes it OK if a lot of lower level plugins depend on it.